### PR TITLE
[chore] Updated `unmarshalText` func

### PR DESCRIPTION
**Pull request by AkhigbeEromo**
Changed the switch statement in` UnmarshalText` function to an if statement {configcompression}
Link to the issue: https://github.com/open-telemetry/opentelemetry-collector/issues/9458

### CODE REVIEW

**Comment by AkhigbeEromo** [comment]
hey @TylerHelmuth, I renamed the branch

Hello @dmitryax can I please get a review

**Review by TylerHelmuth — APPROVED** [review]

File: .chloggen/configcompression-changed-UnmarshalText-func.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [api]
```
[the remark this answers is not in the document]
Comment (AkhigbeEromo) [reply .chloggen/configcompression-changed-UnmarshalText-func.yaml:25]: So i should just delete the file I made in the `.chloggen` folder? @dmitryax

[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply .chloggen/configcompression-changed-UnmarshalText-func.yaml:25]: Yes

[the remark this answers is not in the document]
Comment (AkhigbeEromo) [reply .chloggen/configcompression-changed-UnmarshalText-func.yaml]: Done.
Thank you for the reviews @TylerHelmuth @dmitryax

**Pull request merged**